### ISSUE TITLE
Unable to expose both native and original histograms

### ISSUE THREAD

**Issue by benclapp** [prometheus/statsd_exporter#500]
It doesn't appear to be possible to expose both native, and the original type histograms. There cases where people may want to have some metrics using the native type, and other metrics exposed as the original type, with hard coded buckets.

Additionally, when a Prometheus server enables the native histogram feature flag, all histogram metrics scraped from statsd_exporter will be scraped as native histograms, rather than original histograms.

According to the description of the [Prometheus feature flag](https://prometheus.io/docs/prometheus/latest/feature_flags/#native-histograms), as well as a golang test app below, it should be possible for a scrape target to expose both histogram types.

Example app that exposes histograms of each type, confirming behaviour of client_golang as well.
<details>
  <summary>Expand</summary>

```golang
package main

import (
	"log"
	"math/rand"
	"net/http"
	"time"

	"github.com/prometheus/client_golang/prometheus"
	"github.com/prometheus/client_golang/prometheus/promhttp"
)

func main() {

	registry := prometheus.NewRegistry()

	mUniversal :=
		prometheus.NewHistogram(prometheus.HistogramOpts{
			Name:                           "http_duration_seconds",
			Help:                           "RPC latency distributions.",
			ConstLabels:                    map[string]string{"type": "universal"},
			Buckets:                        []float64{0.01, 0.02, 0.05, 0.1, 0.2, 0.5, 1, 2, 5, 10, 30},
			NativeHistogramBucketFactor:    1.1,
			NativeHistogramMaxBucketNumber: 256,
		})
	mOriginal :=
		prometheus.NewHistogram(prometheus.HistogramOpts{
			Name:        "http_og_duration_seconds",
			Help:        "RPC latency distributions.",
			ConstLabels: map[string]string{"type": "original"},
			Buckets:     []float64{0.01, 0.02, 0.05, 0.1, 0.2, 0.5, 1, 2, 5, 10, 30},
		})
	mNative :=
		prometheus.NewHistogram(prometheus.HistogramOpts{
			Name:                           "http_duration_seconds",
			Help:                           "RPC latency distributions.",
			ConstLabels:                    map[string]string{"type": "native"},
			NativeHistogramBucketFactor:    1.1,
			NativeHistogramMaxBucketNumber: 256,
		})

	registry.MustRegister(mUniversal, mOriginal, mNative)

	go func() {
		for {
			v := rand.NormFloat64()*1 + 0.1
			mUniversal.Observe(v)
			mOriginal.Observe(v)
			mNative.Observe(v)
			time.Sleep(time.Duration(25) * time.Millisecond)
		}
	}()

	http.Handle("/metrics", promhttp.HandlerFor(
		registry,
		promhttp.HandlerOpts{
			EnableOpenMetrics: true,
			Registry:          registry,
		},
	))
	log.Fatal(http.ListenAndServe("0.0.0.0:10000", nil))
}
```
</details>

When the `native_histogram_bucket_factor` and `native_histogram_max_buckets` fields are not set in the mapping config file, they get set to some default values https://github.com/prometheus/statsd_exporter/blob/master/pkg/mapper/mapper.go#L93-L98. Meaning as soon as the native histogram feature flag enabled on the Prometheus server, all scrapes of statsd_exporter where the mapping for timers has been configured to be histograms, will be scraped as native histograms instead. 

Perhaps the default behaviour for the native parameters should be to stay disabled, rather than enabled by default? Or at least disabled by configuration?

**Comment by beorn7** [comment]
client_golang is designed to expose whatever you configure in your instrumentation, and it is very easy to expose both native histograms alongside classic histograms. It is then up to the scraper what to do with it.

Prometheus without the feature flag set ignores native histograms altogether and scrapes only classic histograms. (In case a histogram is only exposed as native histogram, it would only scrape the _count and _sum.)

Prometheus with the feature flag enabled prefers native histograms if they are exposed, and will fall back to classic histograms for histograms that are exposed as only classic histograms.

Furthermore, there is now a [new scrape config option](https://github.com/prometheus/prometheus/pull/12350) to scrape _both_ native and classic histograms where they are exposed simultaneously. It will be part of the upcoming release v2.45.0.

**Comment by matthiasr** [comment]
Oh neat, I did not know about that option, great! I agree that controlling this on the scrape side is easier to scale than adding configuration to every exporter that exposes histograms.

@benclapp just to be clear, the intention in the exporter is to _expose_ both kinds. It sounds to me like your problem is related to the preference during _scraping_ and the new scrape config option will solve that?

**Comment by benclapp** [comment]
The new `scrape_classic_histograms` flag in the scrape config would indeed solve our use case, very exciting!

This does solves our direct problem. It was mostly to keep classic histogram buckets around for various SLO's with specific latency thresholds. And ideally use native histograms for visualisations. 

It's up to interpretation what the behaviour of `statsd_exporter` should be with regards to disabling native histograms - I don't believe there's a way to disable native histograms in `statsd_exporter` if you don't want to expose them, for whatever reason. But they're awesome so can't imagine why 😅 

However I'd be happy to close this issue with a fix for our use case coming in Prometheus 2.45.0. People could always drop native histograms with relabeling if so desired. (A `__meta_metric_type` or similar label would help in this scenario).
